apps/ejabberd/src/mod_privacy_riak.erl

```diff
@@ -0,0 +1,166 @@
+    case get_default_list_name(LUser, LServer) of
+        none ->
+            {error, not_found};
+        Default ->
+            case get_privacy_list(LUser, LServer, Default) of
+                {ok, List} ->
+                    {ok, {Default, List}};
+                {error, Reason} ->
+                    {error, Reason}
+            end
+    end.
+
+get_list_names(LUser, LServer) ->
+    Default = get_default_list_name(LUser, LServer),
+    Names = get_list_names_only(LUser, LServer),
+    {ok, {Default, Names}}.
+
+get_default_list_name(LUser, LServer) ->
+    case mongoose_riak:get(?BKT_DEFAULT_LIST(LServer), LUser) of
+        {ok, Obj} ->
+            riakc_obj:get_value(Obj);
+
+        _ -> none
+    end.
+
+get_list_names_only(LUser, LServer) ->
+    case mongoose_riak:fetch_type(?BKT_LISTS_NAMES(LServer), LUser) of
+        {ok, Set} ->
+            riakc_set:value(Set);
+
+        {error, {notfound, set}} ->
+            [];
+
+        Err -> 
+            lager:error("~p", [Err]),
+            []
+    end.
+
+get_privacy_list(LUser, LServer, Name) ->
+    case mongoose_riak:get(?BKT_LISTS(LServer), <<LUser/binary, $/, Name/binary>>) of
+        {ok, Obj} ->
+            Val = binary_to_term(riakc_obj:get_value(Obj)),
+            {ok, Val};
+
+        {error, notfound} ->
+            {error, not_found};
+
+        Err->
+            lager:error("~p", [Err]),
+            Err
+    end.
+
+forget_default_list(LUser, LServer) ->
+    mongoose_riak:delete(?BKT_DEFAULT_LIST(LServer), LUser).
+
+set_default_list(LUser, LServer, Name) ->
+    case mongoose_riak:get(?BKT_LISTS(LServer), <<LUser/binary, $/, Name/binary>>) of
+        {ok, _} ->
+            case mongoose_riak:get(?BKT_DEFAULT_LIST(LServer), LUser) of
+                % update entry
+                {ok, Obj} ->
+                    Obj2 = riack_obj:update_value(Obj, Name),
+                    mongoose_riak:put(Obj2);
+
+                % create entry
+                {error, notfound} ->
+                    Obj = riakc_obj:new(?BKT_DEFAULT_LIST(LServer), LUser, Name),
+                    mongoose_riak:put(Obj);
+
+                Err ->
+                    lager:error("~p", [Err]),
+                    Err
+            end;
+
+        % in case list name is not found
+        {error, notfound} ->
+            {error, not_found};
+
+        Err -> Err
+    end.
+
+remove_privacy_list(LUser, LServer, Name) ->
+    mongoose_riak:delete(?BKT_LISTS(LServer), <<LUser/binary, $/, Name/binary>>),
+
+    case mongoose_riak:fetch_type(?BKT_LISTS_NAMES(LServer), LUser) of
+        {ok, S1} ->
+            S2 = riakc_set:del_element(Name, S1),
+            mongoose_riak:update_type(?BKT_LISTS_NAMES(LServer), LUser, riakc_set:to_op(S2));
+
+        Err -> 
+            lager:error("~p", [Err]),
+            Err 
+    end.
+
+replace_privacy_list(LUser, LServer, Name, List) ->
+    % store privacy-list content
+    %NOTE: List is automatically serialized
```

Comment:
It is serialized using BERT format. See L89 how I decode read value.
Other clients would require to decode it also.
i.e in python, you can use https://github.com/samuel/python-bert: 

    > http http://localhost:8098/types/privacy_lists/buckets/localhost/keys/jdoe |python3 -u  -c "import os;import sys;import bert; sys.stdin = os.fdopen(sys.stdin.fileno(), 'rb', 0); print(bert.codec.BERTDecoder().decode(sys.stdin.read()))"
    (Atom('foo'), Atom('bar'))